src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -97,25 +113,31 @@ CHIP_ERROR ReadDerUnsignedIntegerIntoRaw(Reader & reader, uint8_t * raw_integer_
     // We now have the rest of the tag that is a "minimal length" unsigned integer.
     // Blit it at the correct offset, since the order we use is MSB first for
     // both ASN.1 and EC curve raw points.
-    size_t offset = raw_integer_length - integer_len;
-    return reader.ReadBytes(&raw_integer_out[offset], integer_len).StatusCode();
+    size_t offset = raw_integer_out.size() - integer_len;
+    return reader.ReadBytes(raw_integer_out.data() + offset, integer_len).StatusCode();
 }
 
-size_t EmitDerIntegerFromRaw(const uint8_t * raw_integer, size_t raw_integer_length_bytes, uint8_t * out_der_integer,
-                             size_t out_der_integer_size)
+/**
+ * @brief Utility to emit a DER-encoded INTEGER given a raw unsigned large integer
+ *        in big-endian order.
+ * @param[in] raw_integer Bytes of a large unsigned integer in big-endian, possibly including leading zeroes
+ * @param[out] out_der_integer Buffer to receive the DER-encoded integer
+ * @return The non-zero length used by DER encoding in `out_der_ingeger`, or 0 on error, such as value too large to fit output.
```

Comment:
We could also just resize `out_der_integer` to have the right size (using `reduce_size`) and have it be an lvalue ref... Either way, I guess.

Reply:
I disagree with `reduce_size` since it's not part of std::span and break my mental model assumption that spans are immutable. If we are going to use "std-like" primitives until C++20 support, I'd rather we not use that mutator.

Reply:
OK.  How do you feel about taking the incoming span by ref and doing `out_span = MutableByteSpan(ptr, newsize)`?  The effect is the same, of course, but does not use `reduce_size`.

The high-level goal here, for me, is to keep the buffer and length together in a single package to avoid confusion about which lengths go with which buffer pointers at what points.  If spans are just temporaries created over a buffer this makes no sense, of course, but if we use a "persistent" span to keep track of "the thing we are working with" like so:
```
MutableByteSpan work_area(myBuffer);
ReturnErrorIfFailure(FillStuff(work_area)); // Takes `MutableByteSpan &` and assigns to it
// Now work_area spans the actual useful data
```
that would make a lot of sense to me...  Thoughts?

Reply:
Applied this comment. It's more logical to me to re-assign than to call `reduce_size`.